docs/source/observations.rst

```diff
@@ -0,0 +1,188 @@
+
+
+labels
+------
+When available (ie when doing supervised learning), this field will contain
+any appropriate labels. For many tasks, this will be only a single response.
+However, some datasets do support multiple correct answers, so this field
+should be an ``iterable`` (e.g. a list, a tuple).
+
+
+label_candidates
+----------------
+For tasks that support ranking, this field is an ``iterable`` of
+possible answers the dataset is suggesting the agent could choose from.
+
+For example, the "mnist-qa" task provides the following label_candidates:
+
+.. code-block:: python
+
+    def label_candidates(self):
+        return [str(x) for x in range(10)] + ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine']
+
+
+text_candidates
+---------------
+This is the partner field to "label_candidates". In response to received
+candidates for the label, the model can choose to return an ``iterable``
+of responses in the order it would rank their relevance to the conversation.
+
+This allows the teacher to grade the model's entire ordering, giving the option
+to compute metrics like ``hits@10`` or ``MRR``.
+
+
+episode_done
+------------
+The "episode_done" flag is used to mark the end of an episode.
+Conversations in ParlAI don't necesarily have more than one exchange, but
+many datasets do.
+
+For example, the WikiMovies dataset only has one:
+
+.. code-block:: python
+
+    {
+        'id': 'wikimovies',
+        'text': 'what movies are about ginger rogers?',
+        'labels': ['Top Hat', 'Kitty Foyle', 'The Barkleys of Broadway'],
+        episode_done=True,
+    }
+
+On the other hand, the bAbI tasks have multiple exchanges per conversation:
+
+.. code-block:: python
+
+    {
+        'id': 'babi:task1k:15',
+        'text': '''Cats are afraid of sheep.
+            Sheep are afraid of mice.
+            Wolves are afraid of sheep.
+            Gertrude is a cat.
+            Winona is a cat.
+            Emily is a sheep.
+            Jessica is a cat.
+            Mice are afraid of cats.
+            What is winona afraid of?''',
+        'labels': ['sheep'],
+        'label_candidates': ['wolf', 'mouse', 'cat', 'sheep'],
+        episode_done=False,
+    }
+    {
+        'id': 'babi:task1k:15',
+        'text': 'What is jessica afraid of?',
+        'labels': ['sheep'],
+        'label_candidates': ['wolf', 'mouse', 'cat', 'sheep'],
+        episode_done=False,
+    }
+    {
+        'id': 'babi:task1k:15',
+        'text': 'What is gertrude afraid of?',
+        'labels': ['sheep'],
+        'label_candidates': ['wolf', 'mouse', 'cat', 'sheep'],
+        episode_done=False,
+    }
+    {
+        'id': 'babi:task1k:15',
+        'text': 'What is emily afraid of?',
+        'labels': ['mouse'],
+        'label_candidates': ['wolf', 'mouse', 'cat', 'sheep'],
+        episode_done=True,
+    }
+
+
+reward
+------
+This field can be used by reinforcement-learning tasks to send rewards to
```

Comment:
'to other agents' is a bit ambiguous, in the observation dict it is just observed so leave it at that i think?